Sherlock/sherlock.h

```diff
@@ -333,26 +337,34 @@ class StreamImpl {
 
   // Asynchronous subscription: `subscriber` is a heap-allocated object, the ownership of which
   // can be `std::move()`-d into the subscriber thread. It can be `Join()`-ed or `Detach()`-ed.
-  template <typename F>
-  using AsyncSubscriberScope = GenericSubscriberScope<F, std::unique_ptr<F>, true>;
-
-  template <typename F>
-  AsyncSubscriberScope<F> Subscribe(std::unique_ptr<F>&& subscriber) {
-    static_assert(current::ss::IsStreamSubscriber<F, ENTRY>::value, "");
-    return std::move(AsyncSubscriberScope<F>(data_, std::move(subscriber)));
+  // The order of two template parameters is flipped to provide the default value to the `SUBSCRIBER_TYPE` one,
```

Comment:
The reason template parameters are flipped is to enable the use of specialized template parameters using the default value.

For `*SubscriberScope`, if the user wants to fully specialie the type instead of using `auto`, the one-template-parameter version would cite user subscriber class as an explicit parameter, assuming the type of entries this listener is subscribed to is the type of entries stored by this Sherlock stream. Thus, it makes sense to keep the type of entries to subscribe to as the second parameter.

At the same time, while creating a `Subscriber`, the type of user class is something which would naturally infer itself, while type filter, if set, must be provided explicitly. Thus, it makes sense to keep `F` as the second template parameter.

I've cleaned this up a bit. PTAL.